build/Targets/Settings.props

```diff
@@ -51,6 +51,16 @@
     <!-- To copy .dll.config and .exe.config files of the referenced projects to the output dir
          (see https://github.com/Microsoft/msbuild/issues/1307) -->
     <AllowedReferenceRelatedFileExtensions>$(AllowedReferenceRelatedFileExtensions);.dll.config;.exe.config</AllowedReferenceRelatedFileExtensions>
+
+    <!-- This controls the places MSBuild will consult to resolve assembly references.  This is 
+         kept as minimal as possible to make our build reliable from machine to machine.  Global
+         locations such as GAC, AssemblyFoldersEx, etc ... are deliberately removed from this 
+         list as they will not be the same from machine to machine -->
+    <AssemblySearchPaths>
+      {CandidateAssemblyFiles};
```

Comment:
@agocke The MSBuild comment for this property is 

> (1) Files from current project - indicated by {CandidateAssemblyFiles}

No idea what that means though.  I deleted it locally and was able to rebuild.  

Would prefer to fix this delete up with a separate PR though.  Want any changes to go through full validation as it can break roslyn-internal in sneaky ways.  Have another PR backed up on this one that I ned to get into teh queue.